python/tchannel/socket.py

```diff
@@ -107,6 +108,12 @@ def close(self):
         self.closed = True
         return self.connection.close()
 
+    def message_and_write(self, message, message_id=None):
```

Comment:
I just checked out your branch and we might have a similar performance degradation in the streaming case. So for now we probably want to do what @jcorbin is doing with node, where we allow both code paths (complex streaming, simplified unstreamed case) until we can figure out how to speed up this path.

Additionally I would make these functions names more representative of what they do, like `frame_and_write` and `frame_and_write_stream`, so it's more obvious when the caller is explicitly asking for streaming. It would be an error to `frame_and_write` a message over 64kb.